src/app/static/src/app/store/root/actions.ts

```diff
@@ -6,8 +6,26 @@ import {LanguageActions} from "../language/language";
 import {changeLanguage} from "../language/actions";
 import i18next from "i18next";
 
-export interface RootActions extends LanguageActions<RootState>{
+export interface RootActions extends LanguageActions<RootState> {
     validate: (store: ActionContext<RootState, RootState>) => void;
+    generateErrorReport: (store: ActionContext<RootState, RootState>,
+                          payload: ErrorReportManualDetails) => ErrorReport;
+}
+
+export interface ErrorReportManualDetails {
+    section: string,
+    description: string,
+    stepsToReproduce: string,
+    email: string
+}
+
+export interface ErrorReport extends ErrorReportManualDetails {
+    country: string,
+    project: string | undefined,
+    browserAgent: string,
+    timeStamp: string,
```

Comment:
timeStamp is generated in the backend automatically, so you won't need to add it in frontend.

Reply:
Agreed - I think it's preferable to add timestamp here, even if backend does intend to fiddle with the JSON.